### PR TITLE
Refactor Blocks/LeadImage/Edit component

### PR DESCRIPTION
**Pull request by Tishasoumya-02**
#4460

### CODE REVIEW

**Review by nileshgulia1 — APPROVED** [review]

**Comment by nileshgulia1** [comment]
@Tishasoumya-02 Can you pls take care of conflicts?

**Review by nileshgulia1 — APPROVED** [review]

**Review by nileshgulia1 — APPROVED** [review]

File: src/components/manage/Blocks/LeadImage/Edit.jsx [hunk 1]
```diff
@@ -23,124 +16,82 @@ const messages = defineMessages({
...
-      nextProps.selected ||
-      !isEqual(this.props.data, nextProps.data)
-    );
-  }
+const Edit = React.memo((props) => {
+  const intl = useIntl();
+  const { data, properties, selected } = props;
 
-  node = React.createRef();
+  const placeholder = useMemo(
```
Comment (iFlameing) [inline src/components/manage/Blocks/LeadImage/Edit.jsx:23]: useMemo is not needed. we are not making significant improvements in performance here.

**Comment by iFlameing** [comment]
@sneridagh I am not able to test it locally because the backend is providing the data. 
@Tishasoumya-02 @nileshgulia1 can you please tell me the steps to see the lead image locally? In this way, I will be sure everything works like old ways. please!

**Comment by nileshgulia1** [comment]
> @sneridagh I am not able to test it locally because the backend is providing the data. @Tishasoumya-02 @nileshgulia1 can you please tell me the steps to see the lead image locally? In this way, I will be sure everything works like old ways. please!

@iFlameing Try to add a image in Lead image behaviour in your content-type and then add a lead image block.

File: packages/volto/src/components/manage/Blocks/LeadImage/Edit.jsx [hunk 2]
```diff
@@ -23,124 +16,79 @@ const messages = defineMessages({
...
-   * @memberof Edit
-   */
-  shouldComponentUpdate(nextProps) {
-    return (
-      this.props.selected ||
-      nextProps.selected ||
-      !isEqual(this.props.data, nextProps.data)
-    );
-  }
+const Edit = React.memo((props) => {
```
Comment (davisagli) [inline packages/volto/src/components/manage/Blocks/LeadImage/Edit.jsx:19]: Remove React.memo

File: packages/volto/src/components/manage/Blocks/LeadImage/Edit.jsx [hunk 3]
```diff
@@ -23,124 +16,79 @@ const messages = defineMessages({
...
-            className={className}
-            item={properties}
-            imageField="image"
-            sizes={(() => {
-              if (data.align === 'full' || data.align === 'center')
-                return '100vw';
-              if (data.align === 'left' || data.align === 'right')
-                return '50vw';
-              return undefined;
-            })()}
```
Comment (davisagli) [inline packages/volto/src/components/manage/Blocks/LeadImage/Edit.jsx:134]: The old code defined a function and called it immediately. We need to do the same thing in the new code. (This is why the snapshot changed, but it should not.)

**Review by davisagli — CHANGES_REQUESTED** [review]

**Review by davisagli — APPROVED** [review]

**Pull request merged**